rllib/algorithms/algorithm.py

```diff
@@ -3413,6 +3413,20 @@ def _compile_iteration_results_new_api_stack(
     ):
         # Return dict (shallow copy of `train_results`).
         results: ResultDict = train_results.copy()
+
+        # TODO (sven): Fix Tune, instead, to be tolerant against possibly missing result
+        #  keys. Otherwise, we'll have to guess here, what "popular" keys users use in
+        #  order to protect them from running into Tune KeyErrors.
+        if ENV_RUNNER_RESULTS not in results:
+            results[ENV_RUNNER_RESULTS] = {}
+        for must_have in [
+            EPISODE_RETURN_MEAN,
+            EPISODE_RETURN_MIN,
+            EPISODE_RETURN_MAX,
+        ]:
+            if must_have not in results[ENV_RUNNER_RESULTS]:
```

Comment:
Yup, understood. But this means, if a user collects custom metrics via callback and uses Tune on this ... an error could result due to this metrics not being available in an iteration. We should document this somewhere - maybe at the MetricsLogger API ...

Reply:
Correct, that's why I put the comment there. We should actually fix Tune, there is no other solution to this problem, imo.

Tune didn't probably think of this b/c in SL, you don't have "strangely behaving" episodes that don't deliver data sometimes.